### PR TITLE
Move away from `material-table` to `@material-table/core`

### LINKED ISSUE #6452: Backstage core Table pagination not working
We have discovered a recent regression in the pagination functionality of the `@backstage/core-components` Table component in our Backstage installation.

## Expected Behavior
When a user presses the "next page" or "previous page" arrow buttons to toggle pages within a Table it should display the next set of data.

## Current Behavior
When a user presses the "next page" or "previous page" arrow buttons no action is performed. An error is thrown and displayed in the console:
Screen Shot 2021-07-13 at 10 25 09 AM

## Possible Solution
This error seems to be related to https://github.com/mbrn/material-table/issues/2943
A potential solution suggested in that issue is to downgrade to an older material-table version until the downstream issue is fixed. There is currently a PR available that should resolve this issue on material-table's side

## Steps to Reproduce
1. Navigate to the Catalog page
2. Click the "next page" arrow button (assuming your catalog has more entities than currently displayed)

## Context
Currently, our users cannot navigate through the pages of data within the catalog (and other custom portions of the app that uses the Backstage Table component). The users can only search for catalog entities or adjust the rows-per-page.

## Your Environment
- NodeJS Version (v12): Node 14
- Operating System and Version (e.g. Ubuntu 14.04): node:14-buster-slim official Docker image
- Browser Information: Chrome Version 91.0.4472.114

### LINKED ISSUE #6553: [TablePagination] Uncaught TypeError: _this.props.onChangePage is not a function
We're using `@material-ui/core@^4.11.0` and experiencing the following error when clicking the button to go the next page:
```
m-table-pagination.js:67 Uncaught TypeError: _this.props.onChangePage is not a function
    at eval (m-table-pagination.js:67)
```

This code previously worked:
```tsx
import { TablePagination } from '@material-ui/core';

export default function Demo() {
  return (
    <TablePagination
      component="div"
      count={50}
      page={0}
      rowsPerPageOptions={[5, 10, 20]}
      rowsPerPage={10}
      onChangeRowsPerPage={(event) => console.log(event.target.value)}
      onChangePage={(_event, newPage) => console.log(newPage)}
    />
  );
}
```

TypeScript complains that `onPageChange` is needed and that `onChangeRowsPerPage` and `onChangePage` are deprecated.

So we changed the code to:
```tsx
<TablePagination
  {/* ... */}
  onRowsPerPageChange={(event) => console.log(event.target.value)}
  onPageChange={(_event, newPage) => console.log(newPage)}
/>
```

With this we're still getting the same error as above.

I tried creating a clean repo with only React and `@material-ui/core` as the dependencies and this error went away.

It would appear that somehow it's using the component from the `material-table` package instead even though we're not importing anything from `material-table`:
```bash
> find . -name "m-table-pagination.js"
./node_modules/material-table/dist/components/m-table-pagination.js
```

## Your Environment

- NodeJS Version: `v14.13.0`
- Operating System and Version: `macOS 10.15.7`
- Browser Information: `Chrome 91.0.4472.114`

### PR DESCRIPTION
**Pull request by benjdlambert**
Signed-off-by: blam <ben@blam.sh>

The `material-table` dependency has become a little stale, and it's under active development. A new fork has emerged https://github.com/material-table-core/core and is being kept up to date.

Fixes #6553 
Fixes #6452 

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by timbonicus — APPROVED** [review]

**Comment by ConnorDY** [comment]
Thank you!!

**Review by joeporpeglia — APPROVED** [review]

**Comment by benjdlambert** [comment]
This is proving a little tricky to get the build to pass as it broke some of the tests - more investigation needed tomorrow not had so much time to look at this today

**Review by joaomilho — APPROVED** [review]

File: .changeset/lazy-colts-help.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+'@backstage/plugin-techdocs': patch
+'@backstage/plugin-todo': patch
+'@backstage/plugin-user-settings': patch
+'@backstage/plugin-welcome': patch
+'@backstage/plugin-xcmetrics': patch
+---
+
+- Bumping `material-ui/core` version to at least `4.12.2` as they made some breaking changes in later versions which broke Pagination of the Table.
+
+- Switching out `material-table` to `@material-table/core` for support for the later versions of `material-ui/core`
```
Comment (dhenneke) [inline .changeset/lazy-colts-help.md:56]: NIT: How about mentioning that the `onChangePage` and `onChangeRowsPerPage` properties have been renamed in the `Table` component?

Comment (benjdlambert) [reply .changeset/lazy-colts-help.md:56]: Good shout, I purposefully left this out as our Table API has stayed the same it's only if using `material-table` directly. But I can add something to the changeset stating this.

Comment (benjdlambert) [reply .changeset/lazy-colts-help.md:56]: I think next release we will add support for both props, and then remove the old one and do a proper deprecation like material-ui should have done :shrug:

Comment (dhenneke) [reply .changeset/lazy-colts-help.md:56]: > Good shout, I purposefully left this out as our Table API has stayed the same it's only if using material-table directly. But I can add something to the changeset stating this.

I'm not sure about that. Didn't you update the tables in the circleci, bitrise, and cloudbuild plugins that all use `import { Table } from '@backstage/core-components`?

Comment (benjdlambert) [reply .changeset/lazy-colts-help.md:56]: Hmm good catch - The `api-reports` just re-exports the props, so when I just scanned to find what I'd done again I didn't see it changing the props. https://github.com/backstage/backstage/pull/6581/files?file-filters%5B%5D=.md#diff-ea569eb8425f004ed49a86194d7b5b19b5dff946ced5c6f77933d48edbb0f036R29

**Review by OrkoHunter — APPROVED** [review]

**Review by timbonicus — APPROVED** [review]

**Pull request merged**